test/restart.jl

```diff
@@ -0,0 +1,297 @@
+import ClimaAtmos as CA
+import ClimaCore
+import ClimaCore: DataLayouts, Fields, Geometry
+import ClimaComms
+pkgversion(ClimaComms) >= v"0.6" && ClimaComms.@import_required_backends
+import Logging
+using Test
+
+const comms_ctx = ClimaComms.context()
+ClimaComms.init(comms_ctx)
+
+# PRINT can be used to turn on more information about differences. It can also
+# be used to only print from root with PRINT = ClimaComms.iamroot(comms_ctx)
+const PRINT = false
+
+# This test checks that:
+
+# 1. A simulation, saved to a checkpoint, is read back identically (up to some
+#   tolerance and excluding those fields that are computed during the
+#   calculation of the tendencies)
+# 2. A simulation, saved to a previous checkpoint, and read back and evolved to
+#   the same time is identical (up to some tolerance)
+# 3. ClimaAtmos can automatically detect restarts
+# 4. It is not possible to change the Atmos model across simulations
+
+"""
+    compare(one, two; name = "", ignore = Set([]))
+
+Recursively compare `one` and `two` up to some numeric tolerance.
+
+`compare` walks through all the properties in `one` and `two` until it finds
+that there are no more properties. At that point, `compare` tries to match the
+resulting objects. When such objects are arrays with floating point, `compare`
+defines a notion of `error` that is the following: when the absolute value is
+less than `100eps(eltype)`, `error = absolute_error`, otherwise it is relative
+error. The `error` is then compared against a tolerance.
+
+Keyword arguments
+=================
+
+- `name` is used to collect the name of the property while we go recursively
+  over all the properties. You can pass a base name.
+- `ignore` is a collection of `Symbol`s that identify properties that are
+  ignored when walking through the tree. This is useful for properties that
+  are known to be different (e.g., `output_dir`).
+"""
+function compare(one, two; name = "", ignore = Set([]))
```

Comment:
Can we not use `one` as a variable name? It's already a julia method.

Reply:
This is a pretty complicated function, can we somehow break it down?

Reply:
Also, just to note, ClimaCore does define `Fields.rcompare` (which is basically used in `==` for Fields).

Reply:
> Also, just to note, ClimaCore does define `Fields.rcompare` (which is basically used in `==` for Fields).

Can I use it for approximate equality?

Reply:
Hm, I don't think so (off hand), but I think it's fewer lines of code if you want to use that pattern instead